### PR TITLE
Open Intake Command and Subsystem

### PR DESCRIPTION
**Pull request by OhadLevine**
This is a command for setting the intake subsystem to be either open or closed

### CODE REVIEW

File: src/main/java/frc/robot/commands/command_groups/CollectCell.java [hunk 1]
```diff
@@ -11,7 +11,7 @@
     public CollectCell() {
         addCommands(
             new SpinMixer(),
-            new MoveMovableSubsystem(intake, () -> robotConstants.intakeConstants.kIntakeDefaultPower)
+            new MoveMovableSubsystem(intake, () -> robotConstants.intakeConstants.kCellIntakeDefaultPower)
         );
```
Comment (Mboesn) [inline src/main/java/frc/robot/commands/command_groups/CollectCell.java:23]: shouldn't open intake be in here

File: src/main/java/frc/robot/constants/RobotConstants.java [hunk 2]
```diff
@@ -37,8 +37,16 @@
...
 
     public static class IntakeConstants {
-        public boolean kIntakeReversed;
-        public double kIntakeDefaultPower;
+        public boolean kCellIntakeReversed;
+        public double kCellIntakeDefaultPower;
+        public boolean kIntakeOpenerReversed;
+        public double kCurrentLimit;
+        public double kThresholdLimit;
+        public double kTriggerThresholdTime;
```
Comment (Mboesn) [inline src/main/java/frc/robot/constants/RobotConstants.java:54]: add these constants for both motors

File: src/main/java/frc/robot/constants/RobotMap.java [hunk 3]
```diff
@@ -28,6 +28,7 @@
         public int kTemporaryTalonForLeftDrivetrainEncoder;
         // Intake ports
         public int kIntakeSparkMax;
+        public int kIntakeTalonSRX;
```
Comment (Mboesn) [inline src/main/java/frc/robot/constants/RobotMap.java]: change the operner motor to kIntakeOpener*motor name*

File: src/main/java/frc/robot/constants/robots/RobotA.java [hunk 4]
```diff
@@ -104,6 +113,8 @@ public RobotA() {
         can.kTemporaryTalonForRightDrivetrainEncoder = 8;
         // Intake Map
         can.kIntakeSparkMax = 11;
+        can.kIntakeTalonSRX = 16;
```
Comment (Mboesn) [inline src/main/java/frc/robot/constants/robots/RobotA.java]: same as in robotmap

File: src/main/java/frc/robot/constants/robots/RobotA.java [hunk 5]
```diff
@@ -30,8 +30,17 @@ public RobotA() {
...
 
         // Intake Constants
-        intakeConstants.kIntakeReversed = false;
-        intakeConstants.kIntakeDefaultPower = 0.3;
+        intakeConstants.kCellIntakeReversed = false;
+        intakeConstants.kCellIntakeDefaultPower = 0.3;
+        intakeConstants.kIntakeOpenerReversed = false;
+        intakeConstants.kCurrentLimit = 0;
+        intakeConstants.kThresholdLimit = 0;
+        intakeConstants.kTriggerThresholdTime = 0;
```
Comment (Mboesn) [inline src/main/java/frc/robot/constants/robots/RobotA.java]: match the new robot constants

File: src/main/java/frc/robot/subsystems/intake/Intake.java [hunk 6]
```diff
@@ -11,18 +16,46 @@
  * This class holds all the methods for the intake which collects POWER CELLS
  * into the robot.
  */
-public class Intake extends OverridableSubsystem {
-    private CANSparkMax sparkMax;
+public class Intake extends OverridableSubsystem implements MovableSubsystem {
+    private CANSparkMax cellIntake;
+    private AnalogPotentiometer potentiometer;
+    private WPI_TalonSRX intakeOpener;
```
Comment (Mboesn) [inline src/main/java/frc/robot/subsystems/intake/Intake.java]: move the talon to be under/above the spark (motor near motor)

File: src/main/java/frc/robot/subsystems/intake/Intake.java [hunk 7]
```diff
@@ -11,18 +16,46 @@
...
 
     public Intake() {
-        sparkMax = new CANSparkMax(robotConstants.can.kIntakeSparkMax, MotorType.kBrushless);
-        sparkMax.setInverted(robotConstants.intakeConstants.kIntakeReversed);
-        sparkMax.setIdleMode(IdleMode.kCoast);
-        sparkMax.burnFlash();
+        cellIntake = new CANSparkMax(robotConstants.can.kIntakeSparkMax, MotorType.kBrushless);
+        cellIntake.setInverted(robotConstants.intakeConstants.kCellIntakeReversed);
+        cellIntake.setIdleMode(IdleMode.kCoast);
+        cellIntake.burnFlash();
```
Comment (Mboesn) [inline src/main/java/frc/robot/subsystems/intake/Intake.java]: if you don't know what it is ask heilllllel or remove it

File: src/main/java/frc/robot/subsystems/intake/Intake.java [hunk 8]
```diff
@@ -11,18 +16,46 @@
...
+        intakeOpener.configSupplyCurrentLimit(new SupplyCurrentLimitConfiguration(true,
+                robotConstants.intakeConstants.kCurrentLimit, robotConstants.intakeConstants.kThresholdLimit,
+                robotConstants.intakeConstants.kTriggerThresholdTime));
+
+        potentiometer = new AnalogPotentiometer(robotConstants.pwm.kIntakePotentiometer,
+                robotConstants.intakeConstants.kPotentiometerAngleMultiplier,
+                robotConstants.intakeConstants.kPotentiometerOffset);
+    }
+
+    public double getAngle() {
```
Comment (Mboesn) [inline src/main/java/frc/robot/subsystems/intake/Intake.java]: is parallel to the floor 0 or when it is closed?

File: src/main/java/frc/robot/subsystems/intake/Intake.java [hunk 9]
```diff
@@ -11,18 +16,46 @@
...
+        cellIntake.set(power);
+    }
+
+    /**
+     * Moves the intakeOpener motors
+     */
+    @Override
+    public void move(double power) {
+        intakeOpener.set(power);
     }
```
Comment (Mboesn) [inline src/main/java/frc/robot/subsystems/intake/Intake.java]: why is one move and the other override??? they should be in functions that the name tells the user what it is

**Review by Mboesn — DISMISSED** [review]
fix these before hiellllel can do code review.

File: src/main/java/frc/robot/constants/robots/RobotA.java (hunk 4 above)
Comment (OhadLevine) [reply src/main/java/frc/robot/constants/robots/RobotA.java]: What does this mean?

File: src/main/java/frc/robot/constants/RobotConstants.java (hunk 2 above)
Comment (OhadLevine) [reply src/main/java/frc/robot/constants/RobotConstants.java:54]: cool beans

File: src/main/java/frc/robot/constants/robots/RobotA.java (hunk 4 above)
Comment (OhadLevine) [reply src/main/java/frc/robot/constants/robots/RobotA.java]: yupeedoodoo

**Comment by OhadLevine** [comment]
[Add intake open method with motor](https://trello.com/c/1tE4VErV/146-add-intake-open-method-with-motor)

**Comment by EliyaFishman** [comment]
I think it's a better idea to use ProfiledPIDController than a normal PID

**Comment by HilelAmiel** [comment]
Analog Potentiometer is connected to Roborio Analog Channels, not DIO!
see port 16 in this image:
image

**Review by HilelAmiel — APPROVED** [review]

**Pull request merged**